src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -77,30 +77,48 @@
 
-#define BATT_SMBUS_I2C_BUS			PX4_I2C_BUS_EXPANSION
-#define BATT_SMBUS_ADDR				0x0B	///< I2C address
-#define BATT_SMBUS_TEMP				0x08	///< temperature register
-#define BATT_SMBUS_VOLTAGE			0x09	///< voltage register
+#define BATT_SMBUS_I2C_BUS              PX4_I2C_BUS_EXPANSION
+#define BATT_SMBUS_ADDR                 0x0B	///< I2C address
+#define BATT_SMBUS_TEMP                 0x08	///< temperature register
+#define BATT_SMBUS_VOLTAGE              0x09	///< voltage register
 #define BATT_SMBUS_REMAINING_CAPACITY	0x0f	///< predicted remaining battery capacity as a percentage
 #define BATT_SMBUS_FULL_CHARGE_CAPACITY 0x10    ///< capacity when fully charged
-#define BATT_SMBUS_DESIGN_CAPACITY		0x18	///< design capacity register
-#define BATT_SMBUS_DESIGN_VOLTAGE		0x19	///< design voltage register
-#define BATT_SMBUS_SERIALNUM			0x1c	///< serial number register
-#define BATT_SMBUS_MANUFACTURE_NAME		0x20	///< manufacturer name
-#define BATT_SMBUS_MANUFACTURE_INFO		0x25	///< cell voltage register
-#define BATT_SMBUS_CURRENT			0x2a	///< current register
-#define BATT_SMBUS_MEASUREMENT_INTERVAL_MS	(1000000 / 10)	///< time in microseconds, measure at 10hz
-#define BATT_SMBUS_TIMEOUT_MS		10000000	///< timeout looking for battery 10seconds after startup
-
-#define BATT_SMBUS_PEC_POLYNOMIAL		0x07	///< Polynomial for calculating PEC
+#define BATT_SMBUS_DESIGN_CAPACITY	0x18	///< design capacity register
+#define BATT_SMBUS_DESIGN_VOLTAGE	0x19	///< design voltage register
+#define BATT_SMBUS_MANUFACTURE_DATE   0x1B  ///< manufacture date register
+#define BATT_SMBUS_SERIAL_NUMBER      0x1C  ///< serial number register
+#define BATT_SMBUS_MANUFACTURER_NAME	0x20	///< manufacturer name
+#define BATT_SMBUS_DEVICE_NAME        0x21  ///< device name register
+#define BATT_SMBUS_DEVICE_CHEMISTRY   0x22  ///< device chemistry register
+#define BATT_SMBUS_MANUFACTURER_DATA		0x23	///< manufacturer data
+#define BATT_SMBUS_MANUFACTURE_INFO	0x25	///< cell voltage register
+#define BATT_SMBUS_CURRENT              0x2a	///< current register
+#define BATT_SMBUS_MEASUREMENT_INTERVAL_US	(1000000 / 10)	///< time in microseconds, measure at 10Hz
+#define BATT_SMBUS_TIMEOUT_US			10000000	///< timeout looking for battery 10seconds after startup
+
+#define BATT_SMBUS_BUTTON_DEBOUNCE_MS	300		///< button holds longer than this time will cause a power off event
+
+#define BATT_SMBUS_MANUFACTURER_ACCESS	0x00
+#define BATT_SMBUS_MANUFACTURER_BLOCK_ACCESS    0x44
+
+#define BATT_SMBUS_PEC_POLYNOMIAL	0x07	///< Polynomial for calculating PEC
 
 #ifndef CONFIG_SCHED_WORKQUEUE
 # error This requires CONFIG_SCHED_WORKQUEUE.
 #endif
 
+struct battery_type {
+	char *ManufacturerName;
+	char *DeviceName;
+	char *DeviceChemistry;
+};
+
+// Declaration of the solo battery data, as determined by reading out the data from multiple 3DR Solo batteries
+const struct battery_type solo_battery = {(char *)"BMTPOW", (char *)"MA03", (char *)"LIon"};
```

Comment:
This defines the characteristics or expected 'properties' for a Solo battery against which the Manufacturer Name, Device Name and Device Chemistry is checked when it is read out from the SMBus .